volume/ceph/ceph.go

```diff
@@ -177,24 +186,64 @@ func (v *Volume) Mount(id string) (mappedDevicePath string, returnedError error)
 	if err != nil {
 		return "", err
 	}
+
+	mappedDevicePath = v.mappedDevicePath
+
+	if fsType == cryptoLUKS {
+		cmd = exec.Command("cryptsetup", "luksOpen", "--key-file=-", v.mappedDevicePath, v.Name())
+		stdin, err := cmd.StdinPipe()
+		if err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+		defer stdin.Close()
+		if err := cmd.Start(); err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+
+		key, err := getLuksKey(v.Name())
+		if err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+
+		io.WriteString(stdin, key)
+		stdin.Close()
+		if err := cmd.Wait(); err != nil {
+			logrus.Errorf("Failed to luksOpen Ceph volume '%s' (device %s) - %s", v.Name(), v.mappedDevicePath, err)
+			return "", err
+		}
+
+		v.mappedLuksDevicePath = path.Join(LuksDevMapperPath, v.Name())
+		mappedDevicePath = v.mappedLuksDevicePath
+
+		fsType, err = utils.DeviceHasFilesystem(mappedDevicePath)
+		logrus.Errorf("Checked filesystem in %s: %s", mappedDevicePath, fsType)
+		if err != nil {
+			return "", err
+		}
+	}
+
 	if fsType == "" {
```

Comment:
We want a straight if; If the image is luks encrypted, but does NOT have a filesystem, we want to create a filesystem after we lukeOpen the image.